manifests/service.pp

```diff
@@ -16,11 +16,13 @@
       create_resources(docker::run, $container, $defaults)
     }
     'package','repo': {
-      service { 'grafana':
-        ensure    => running,
-        name      => $grafana::service_name,
-        enable    => true,
-        subscribe => Package['grafana'],
+      if !defined(Service['grafana']) {
```

Comment:
I'd avoid `defined()` as it depends on the evaluation order. Better add another class parameter `$manage_service` and behave accordingly to the parameter value.

Reply:
See, how it's done here e.g.: https://github.com/voxpupuli/puppet-caddy/blob/master/manifests/init.pp#L192